src/runtime-rs/crates/hypervisor/src/qemu/qmp.rs

```diff
@@ -467,8 +473,111 @@ impl Qmp {
 
         Ok(())
     }
+
+    pub fn get_device_by_qdev_id(&mut self, qdev_id: &str) -> Result<PciPath> {
+        let format_str = |vec: &Vec<i64>| -> String {
+            vec.iter()
+                .map(|num| format!("{:02x}", num))
+                .collect::<Vec<String>>()
+                .join("/")
+        };
+
+        let mut path = vec![];
+        let pci = self.qmp.execute(&qapi_qmp::query_pci {})?;
+        for pci_info in pci.iter() {
+            if let Some(_device) = get_pci_path_by_qdev_id(&pci_info.devices, qdev_id, &mut path) {
+                let pci_path = format_str(&path);
+                return PciPath::try_from(pci_path.as_str());
+            }
+        }
+
+        Err(anyhow!("no target device found"))
+    }
+
+    pub fn hotplug_vfio_device(&mut self, vfiodev: &mut VfioDevice) -> Result<()> {
```

Comment:
The function does not deal with a mediated device, does it? I was wondering if it is planned. Thanks.

Reply:
The `vfio` arg introduces a dependency of this module on `crate::VfioDevice`.  Could avoid this by passing the primary device, bus and driver type instead?

Reply:
fixed

Reply:
> The function does not deal with a mediated device, does it? I was wondering if it is planned. Thanks.
>
IMO, currently it focus on the generic `vfio-pci` device via qmp, temporarily excluding mediated device in this PR.